lib/grape/jsonapi/services/content_negotiator.rb

```diff
@@ -0,0 +1,41 @@
+module Grape
+  module Jsonapi
+    module Services
+      class ContentNegotiator
+        VALID_MEDIA_TYPE = 'application/vnd+json'.freeze
+
+        def initialize(accept_header:, content_type:)
+          @accept_header  = accept_header
+          @content_type   = content_type
+        end
+
+        def self.run(*args)
+          new(*args).run
+        end
+
+        def run
+          raise unsupported_media_type_error unless valid?(content_type)
+          raise not_acceptable_error unless valid?(accept_header)
+
+          true
+        end
+
+        private
+
+        attr_reader :accept_header, :content_type
+
+        def valid?(header)
+          VALID_MEDIA_TYPE == header || header.nil?
```

Comment:
Sorry, should've asked this before... So no header is valid ?

Reply:
Ok so it looks like when the Client sends data (`PUT` and `POST`), we require a media type.  So, any way we can check for the type of request and ensure that `header` is not nil if it's a `PUT` or `POST` ?  Is that in the scope of this project ? @chandeeland

Reply:
😆  http://weknowmemes.com/generator/uploads/generated/g1366824328707416474.jpg